forc-pkg/src/pkg.rs

```diff
@@ -163,6 +163,13 @@ pub struct BuildPlan {
     graph: Graph,
     path_map: PathMap,
     compilation_order: Vec<NodeIx>,
+    /// The differences between graph described by lock_file and the graph that actually constructed.
+    ///
+    /// If the BuildPlan is constructed from a manifest file, this will be empty.
+    ///
+    /// If the BuildPlan is constructed from a lock file there might be some differences between the graph described by the lock file and the one that actually created.
+    /// The differences between them are the path dependencies that are removed from the manifest file after the lock file is created.
+    removed_deps: Vec<DependencyName>,
```

Comment:
I am not happy to put a `removed_deps` field in the BuildPlan but couldn't find a cleaner way to pass the `removed_deps` vec from `graph_to_path_map` to validate. So any input would be highly appreciated 🙌

Reply:
Perhaps rather than storing this in the `BuildPlan` itself, we can return this in a tuple alongside `BuildPlan::from_lock`?